### DISCUSSION TITLE
Veeam SOS API

### DISCUSSION THREAD

**Discussion by kevinb0754** [minio/minio#18056] [Q&A] [answered]
Hi,

I have a new MinIO cluster deployed (2023-09-04). I have a bucket created with a hard quota set of 100GB.
I have added this to my Veeam backup server and run a test backup that consumed 20GB of data.

I don't see the Veeam SOS API files being created in the bucket. Is there something I need to do to get this enabled and working?
Because of this Veeam cannot tell the usage of the bucket and lists it as S3 compatible instead of S3 integrated.

Thanks
Kevin

**Comment by harshavardhana** [comment, marked as answer]
> Hi,
> 
> I have a new MinIO cluster deployed (2023-09-04). I have a bucket created with a hard quota set of 100GB. I have added this to my Veeam backup server and run a test backup that consumed 20GB of data.
> 
> I don't see the Veeam SOS API files being created in the bucket. Is there something I need to do to get this enabled and working? Because of this Veeam cannot tell the usage of the bucket and lists it as S3 compatible instead of S3 integrated.
> 
> Thanks Kevin

Veeam SOS files are virtual and available always. There is nothing that will be created on the disk on MinIO related to SOS files.

**Reply by kevinb0754** [reply]
Would I be able to see them from the S3 object browser or are they only available to the Veeam server?

**Reply by harshavardhana** [reply]
They are only available to Veeam @kevinb0754

**Reply by klauspost** [reply]
`mc cat ALIAS/BUCKET/.system-d26a9498-cb7c-4a87-a44a-8ae204f5ba6c/system.xml` will show it.

**Reply by klauspost** [reply]
`mc cat ALIAS/BUCKET/.system-d26a9498-cb7c-4a87-a44a-8ae204f5ba6c/capacity.xml` for capacity

**Reply by kevinb0754** [reply]
Thank you for the command.

mc cat ALIAS/BUCKET/.system-d26a9498-cb7c-4a87-a44a-8ae204f5ba6c/system.xml looks as expected.

mc cat ALIAS/BUCKET/.system-d26a9498-cb7c-4a87-a44a-8ae204f5ba6c/capacity.xml is showing me the capacity, available and used storage for the entire MinIO cluster not just the bucket I am querying.

Is that expected behavior or is there something wrong with my bucket configuration?

**Reply by harshavardhana** [reply]
Yes it is expected.

**Reply by kevinb0754** [reply]
Thank you for the feedback.

Is there anyway to disable this function so as not to report the capacity to all users accessing the same MinIO cluster?
Or is there a way to only report capacity for the bucket being queried?

**Reply by harshavardhana** [reply]
We can add capacity of the bucket only that is being queried @kevinb0754

**Reply by kevinb0754** [reply]
Hi,

Sorry not sure I understand. Is there an option to report on the bucket capacity only?
At the moment it is reporting on the entire cluster capacity not just the bucket.

**Reply by klauspost** [reply]
No there is not. Harsha is suggesting we may add it. Feel free to reach out on `hello@min.io` if you want to discuss further.

**Reply by kevinb0754** [reply]
Just to explain what I am seeing.

I have a bucket1 configured with a 100GB quota and the usage in the bucket is 17GB.
I have a bucket2 configured with a 10GB quota and the usage in the bucket is 0GB.

When I check the capacity.xml for both buckets

mc cat myminio/bucket1/.system-d26a9498-cb7c-4a87-a44a-8ae204f5ba6c/capacity.xml
mc cat myminio/bucket2/.system-d26a9498-cb7c-4a87-a44a-8ae204f5ba6c/capacity.xml

I see both results returning
Capacity 523TB
Available 520TB
Used 3TB

I was hoping to see capacity.xml for bucket1
Capacity 100GB
Available 83GB
Used 17GB

And capacity.xml for bucket2
Capacity 10GB
Available 10GB
Used 0GB

Is there anyway to achieve this output?

**Reply by kevinb0754** [reply]
Ok thank you.

**Reply by kevinb0754** [reply]
Is there anyway to turn this feature off?

**Reply by kevinb0754** [reply]
So just to confirm something. I received a reply from hello@min.io saying that Veeam is writing these files and MinIO is simply storing this for Veeam.

My understanding is that Veeam would not be able to request the capacity of the storage or bucket from MinIO because these functions don't exist in the S3 protocol and that is the reason Veeam developed the SOS API so that this data could be made available.

How then is Veeam writing the files with the capacity data when it doesn't know what that is or am I misunderstanding the this?

Who do I need to send this feature request too, MinIO or Veeam?

**Comment by kevinb0754** [comment]
> Is there anyway to turn this feature off?

Never mind I have blocked access to these files using a policy.
